### ISSUE TITLE
Error while compiling yolo_console_dll.cpp

### ISSUE THREAD

**Issue by karanmartian** [AlexeyAB/darknet#1130]
Hi. I get this error while compiling the cpp

(tf) karan@karan-LinuxTP:~/al/darknet/src$ g++ yolo_console_dll.cpp
/tmp/cc4RfFu5.o: In function `main':
yolo_console_dll.cpp:(.text+0x60b): undefined reference to `Detector::Detector(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int)'
yolo_console_dll.cpp:(.text+0x734): undefined reference to `Detector::load_image(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)'
yolo_console_dll.cpp:(.text+0x77c): undefined reference to `Detector::detect(image_t, float, bool)'
yolo_console_dll.cpp:(.text+0x79b): undefined reference to `Detector::free_image(image_t)'
yolo_console_dll.cpp:(.text+0x85b): undefined reference to `Detector::~Detector()'
yolo_console_dll.cpp:(.text+0xad4): undefined reference to `Detector::~Detector()'
collect2: error: ld returned 1 exit status

**Comment by AlexeyAB** [comment]
Hi,

Did you compile `yolo_cpp_dll.sln` first?

**Comment by karanmartian** [comment]
Actually I am trying to run this on Linux @AlexeyAB. I believe the sln is for VS Studio on Windows

**Comment by AlexeyAB** [comment]
Did you compile darknet with `LIBSO=1` in the Makefile?
And did you run `uselib` in such a way?
`LD_LIBRARY_PATH=./:$LD_LIBRARY_PATH ./uselib test.mp4`

https://github.com/AlexeyAB/darknet#how-to-compile-on-linux

> LIBSO=1 to build a library darknet.so and binary runable file uselib that uses this library. Or you can try to run so LD_LIBRARY_PATH=./:$LD_LIBRARY_PATH ./uselib test.mp4 How to use this SO-library from your own code - you can look at C++ example: https://github.com/AlexeyAB/darknet/blob/master/src/yolo_console_dll.cpp

**Comment by karanmartian** [comment]
@AlexeyAB yes I compiled with LIBSO=1.I have the .so library in place after running make. But I don’t understand how uselib is related to yolo_console_dll.cpp. All i am doing is compiling such a way:
g++ yolo_console_dll.cpp. The code in this .cpp file just references what it needs from the built .so library right? then what is the need of uselib and what does it do?

**Comment by karanmartian** [comment]
@AlexeyAB is that uselib binary the compiled version of yolo_console_dll.cpp? that would explain everything

**Comment by AlexeyAB** [comment]
>  is that uselib binary the compiled version of yolo_console_dll.cpp? that would explain everything

Yes.

----

Look how `uselib` (that contain `yolo_console_dll.cpp`) can be compiled:

https://github.com/AlexeyAB/darknet/blob/6d2e31bf2026301a25a3e8dabeae0680c159328b/Makefile#L122-L123

Where are: https://github.com/AlexeyAB/darknet/blob/6d2e31bf2026301a25a3e8dabeae0680c159328b/Makefile#L44-L45
